src/pudl/output/ferc1.py

```diff
@@ -3044,25 +3084,112 @@ def out_ferc1__yearly_rate_base(
         # the assets/liabilites both use ending_balance for its main $$ column
         .rename(columns={"dollar_value": "ending_balance"})
     )
-    # then select only the leafy exploded records that are in rate base and concat
-    in_rate_base = pd.concat(
+    return cash_working_capital
+
+
+def breakdown_unlabeled(
+    rate_base: pd.DataFrame,
+    split_col: str,
+) -> pd.DataFrame:
+    """Breakdown rate base records with an unlabeled split_col.
+
+    Args:
+        rate_base: full table of rate base data.
+        split_col: column with the label that contains unlabeled values
+            (ex: null or total).
+    """
+    # the works for both the util type and the in rate base tags because
+    # for both tags: unlabeled is represented as a null and but just util type:
+    # unlabeled is also represented as "total"
+    unlabeled_mask = (rate_base[split_col] == "total") | rate_base[split_col].isnull()
+    ratio_idx = ["report_year", "utility_id_ferc1"]
+    ratio_df = get_split_col_ratio(
+        # remove the unlabeled records because total is the value
+        # we want to breakdown so we can't have it in the columns to sum up
+        rate_base[~unlabeled_mask],
+        ratio_idx=ratio_idx,
+        split_col=split_col,
+    )
+    rate_base_broken_down = apply_ratio_to_breakdown_unlabeled(
+        rate_base=rate_base,
+        ratio_df=ratio_df,
+        ratio_idx=ratio_idx,
+        unlabeled_mask=unlabeled_mask,
+        split_col=split_col,
+    )
+    return rate_base_broken_down.convert_dtypes()
+
+
+def get_split_col_ratio(rate_base_labeled, ratio_idx, split_col) -> pd.DataFrame:
+    """Calculate the percentage of the total labeled ``ending_balance`` within each tag group.
+
+    Make ratio column with a 0-1 value of the sum of ``ending_balance`` in each labled
+    ``split_col`` within ``ratio_idx``.
+    """
```

Comment:
i had a hard time explaining these steps..

Reply:
It might be helpful to explain more of the context in the description?

Reply:
yes your questions and suggestions were very helpful to nudge me to clarify what is going on in here. I updated all of the docs.